votingpool/factory_test.go

```diff
@@ -0,0 +1,384 @@
+// Series will use a 2-of-3 configuration and will be empowered with all of its
+// private keys.
+func TstCreateCredits(
+	t *testing.T, pool *Pool, amounts []int64, store *txstore.Store) (uint32, []Credit) {
+	masters := []*hdkeychain.ExtendedKey{
+		TstCreateMasterKey(t, bytes.Repeat(uint32ToBytes(getUniqueID()), 4)),
+		TstCreateMasterKey(t, bytes.Repeat(uint32ToBytes(getUniqueID()), 4)),
+		TstCreateMasterKey(t, bytes.Repeat(uint32ToBytes(getUniqueID()), 4)),
+	}
+	def := TstCreateSeriesDef(t, pool, 2, masters)
+	TstCreateSeries(t, pool, []TstSeriesDef{def})
+	return def.SeriesID, TstCreateCreditsOnSeries(t, pool, def.SeriesID, amounts, store)
+}
+
+// TstCreateCreditsOnSeries creates a slice of credits locked to the given
+// series' address with branch==1 and index==0.
+func TstCreateCreditsOnSeries(t *testing.T, pool *Pool, seriesID uint32, amounts []int64,
+	store *txstore.Store) []Credit {
+	branch := Branch(1)
+	idx := Index(0)
+	pkScript := TstCreatePkScript(t, pool, seriesID, branch, idx)
+	eligible := make([]Credit, len(amounts))
+	for i, credit := range TstCreateInputs(t, store, pkScript, amounts) {
+		eligible[i] = newCredit(credit, *TstNewWithdrawalAddress(t, pool, seriesID, branch, idx))
+	}
+	return eligible
+}
+
+// TstCreateInputs is a convenience function.  See TstCreateInputsOnBlock
+// for a more flexible version.
+func TstCreateInputs(t *testing.T, store *txstore.Store, pkScript []byte, amounts []int64) []txstore.Credit {
+	return TstCreateInputsOnBlock(t, store, 1, pkScript, amounts)
+}
+
+// TstCreateInputsOnBlock creates a number of inputs by creating a transaction
+// with a number of outputs corresponding to the elements of the amounts slice.
+//
+// The transaction is added to a block and the index and blockheight must be
+// specified.
+func TstCreateInputsOnBlock(t *testing.T, s *txstore.Store,
+	blockTxIndex int, pkScript []byte, amounts []int64) []txstore.Credit {
+	msgTx := createMsgTx(pkScript, amounts)
+	block := &txstore.Block{
+		Height: TstInputsBlock,
+	}
+
+	tx := btcutil.NewTx(msgTx)
+	tx.SetIndex(blockTxIndex)
+
+	r, err := s.InsertTx(tx, block)
+	if err != nil {
+		t.Fatal("Failed to create inputs: ", err)
+	}
+
+	credits := make([]txstore.Credit, len(msgTx.TxOut))
+	for i := range msgTx.TxOut {
+		credit, err := r.AddCredit(uint32(i), false)
+		if err != nil {
+			t.Fatal("Failed to create inputs: ", err)
+		}
+		credits[i] = credit
+	}
+	return credits
+}
+
+// TstCreatePool creates a Pool on a fresh walletdb and returns it. It also
+// returns the pool's waddrmgr.Manager (which uses the same walletdb, but with a
+// different namespace) as a convenience, and a teardown function that closes
+// the Manager and removes the directory used to store the database.
+func TstCreatePool(t *testing.T) (tearDownFunc func(), mgr *waddrmgr.Manager, pool *Pool) {
+	// XXX: This should be moved somewhere else eventually as not all of our
```

Comment:
You mean TestMain()? I think that wouldn't work because in that function we don't have access to the testing.T instance. I've changed the comment from an XXX to a regular one since we don't seem to have any better alternatives at this point